project-base/storefront/utils/productLists/useProductList.ts

```diff
@@ -40,6 +41,10 @@ export const useProductList = (
         pause: (!productListUuid && !isUserLoggedIn) || authLoading !== null,
     });
 
+    useEffect(() => {
+        setIsProductListFetching(fetching);
+    }, [fetching]);
```

Comment:
I understand your motivation and thinking here. However, I must say I am not a big fan of this approach. This useEffect and state is a separate logic from the query hook and it actually does not fully match the `fetching` variable above. If it would, you would not be able to use it like this, because it would cause the same hydration problem. Because of that, I would be more in favour of using the same approach as we do in cart (with hydration). This way you postpone the fetching of product lists, but still use the correct fetching variable.

<img width="642" alt="image" src="https://github.com/shopsys/shopsys/assets/55181354/77f56a50-3b63-4483-89f8-252a482bafcb">

Reply:
Ok thank you for the recommendation. I will adjust it.